react/week2/src/App.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import Modal from './components/Modal'
+import Header from './components/Header';
+import AddRandom from './components/AddRandom';
+import "./App.css";
+
+function Todo({ todo, index, completeTodo, removeTodo }) {
+  return (
+    <div
+      className="todo"
+      style={{ textDecoration: todo.isCompleted ? "line-through" : "" }}
+    >
+      {todo.text}
+
+      <div>
+        <button type="checkBox" onClick={() => completeTodo(index)}>Complete</button>
+        <button onClick={() => removeTodo(index)}>Delete</button>
+       </div>
+     
+    </div>
+    
+  );
+}
+
+function TodoForm({ addTodo }) {
+  const [value, setValue] = useState("");
+
+  const handleSubmit = e => {
+    e.preventDefault();
+   
+    if (!value) return;
+    addTodo(value);
+    setValue("");
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        type="text"
+        className="input"
+        placeholder="Enter activity here.."
+        value={value}
+        onChange={e => setValue(e.target.value)}
+      />
+    </form>
+  );
+}
+
+function App() {
+  const [todos, setTodos] = useState([
+    {
+      text: "Get out of bed",
+      isCompleted: false
+    },
+    {
+      text: "Brush teeth",
+      isCompleted: false
+    },
+    {
+      text: "Eat breakfast",
+      isCompleted: true
+    }
+  ]);
+
+  const addTodo = text => {
+    const newTodos = [...todos, { text }];
+    setTodos(newTodos);
+  };
+
+  const completeTodo = index => {
```

Comment:
This function marks todo as completed, and this state can't be changed afterwards. If it wasn't your intention, you can change the logic here to invert isCompleted, instead of setting it to true:

```
newTodos[index].isCompleted = !newTodos[index].isCompleted
```